src/box/watcher.c

```diff
@@ -175,6 +178,8 @@ watchable_schedule_node(struct watchable *watchable,
 			struct watchable_node *node)
 {
 	assert(node->watchable == watchable);
+	if (watchable->is_shutdown)
+		return;
```

Comment:
Instead of adding this check here, better use `watchable_wakeup_worker` below instead of `fiber_wakeup`.

Reply:
Done.

```diff
--- a/src/box/watcher.c
+++ b/src/box/watcher.c
@@ -178,8 +178,6 @@ watchable_schedule_node(struct watchable *watchable,
                        struct watchable_node *node)
 {
        assert(node->watchable == watchable);
-       if (watchable->is_shutdown)
-               return;
        /*
         * Always append to the list tail to guarantee that all watchers
         * eventually run no matter how often nodes are updated.
@@ -187,7 +185,7 @@ watchable_schedule_node(struct watchable *watchable,
        if (!rlist_empty(&node->idle_watchers)) {
                rlist_splice_tail(&watchable->pending_watchers,
                                  &node->idle_watchers);
-               fiber_wakeup(watchable->worker);
+               watchable_wakeup_worker(watchable);
        }
 }
```